src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java

```diff
@@ -741,15 +713,10 @@ public ArtifactVersions lookupArtifactUpdates( Artifact artifact, boolean allowS
                 dependencyUpdates.put( dav.getDependency(), dav.getArtifactVersions() );
             }
         }
-        catch ( final ExecutionException ee )
-        {
-            throw new ArtifactMetadataRetrievalException( "Unable to acquire metadata for dependencies " + dependencies
-                + ": " + ee.getMessage(), ee );
-        }
-        catch ( final InterruptedException ie )
+        catch ( ExecutionException | InterruptedException ie )
         {
             throw new ArtifactMetadataRetrievalException( "Unable to acquire metadata for dependencies " + dependencies
-                + ": " + ie.getMessage(), ie );
+                + ": " + ie.getMessage(), ie, null );
```

Comment:
Change to `org.apache.maven.repository.legacy.metadata.AMRE`.
We might want to extract the current Artifact into a variable outside the try, so we actually know the Artifact  which causes the issue.

Reply:
did you look at context of this? please do so you will realise except very complicated/useless piece of code